src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -1094,6 +1206,22 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
     }.toVector
   }
 
+  /** Merges two sequences of environment variables, with values from secondSeq taking precedence
+    * for duplicate names
+    *
+    * @param firstSeq
+    *   First sequence of environment variables
+    * @param secondSeq
+    *   Second sequence of environment variables
+    * @return
+    *   Merged sequence with duplicates resolved from secondSeq
+    */
+  private[spark] def mergeEnvVars(firstSeq: Seq[EnvVar], secondSeq: Seq[EnvVar]): Seq[EnvVar] = {
+    val secondMap = secondSeq.map(env => env.name -> env).toMap
+    firstSeq.map(env => secondMap.getOrElse(env.name, env)) ++
+      secondSeq.filterNot(env => firstSeq.exists(_.name == env.name))
+  }
```

Comment:
Operation `Map.++` implements the expected precedence:
```suggestion
  private[spark] def mergeEnvVars(firstSeq: Seq[EnvVar], secondSeq: Seq[EnvVar]): Seq[EnvVar] = {
    var index: EnvVar => (String, EnvVar) = env => env.name -> env
    val merged = secondSeq.map(index).toMap ++ firstSeq.map(index).toMap
    merged.values.toSeq
  }
```